### DISCUSSION TITLE
ssh access&public keys

### DISCUSSION THREAD

**Discussion by agorkiy** [gravitational/teleport#13879] [Q&A] [answered]
Hi
I'm looking for simple&easy solution for ssh access managing which could be installed on ssh bastion and be the single access point to nodes behind it. I tried teleport  - it works great, but I have few questions now.
1. Is it possible in some way to just use ssh public keys from users as auth method, instead of password&otp? 
2. If I need restrict users from accessing some nodes - I need:
  1.assign label to selected nodes
  2.create role which applies to that label
  3.create user and attach role to it.
  4.manually dump user definition with ```tctl get``` , edit file - add traits to it and then apply again.(as described [here](https://github.com/gravitational/teleport/discussions/13669))
  Any chances that this last step with ```tctl``` could be done in web admin ui somehow?Or maybe in the future? Cause it looks a bit complicated for non-admin user which may manage teleport accesses so it would be cool if creating user/giving it access to nodes/ssh could be done with some easy clicking in web UI.

**Comment by webvictim** [comment, marked as answer]
1. Teleport is a solution based entirely on short-lived certificates and as such doesn't support the use of SSH public keys. Password and OTP/Webauthn as a second factor is more secure, and only requires the user to log in once per day to get their certificate.

2. You shouldn't need to edit the traits to add a role to a user - you can already do this via the web UI:

![Screenshot 2022-06-27 at 09 55 58](https://user-images.githubusercontent.com/897821/175946984-f24ab85b-ac4f-48e1-86dd-9ed1efa0d4fb.png)

![Screenshot 2022-06-27 at 09 56 23](https://user-images.githubusercontent.com/897821/175947006-a6af2612-6084-4560-8583-e056ac827d90.png)

![Screenshot 2022-06-27 at 09 56 49](https://user-images.githubusercontent.com/897821/175947058-4dcf121c-b44e-492e-a092-40ac9123cc9d.png)

To add an individual username to an individual user then yes, you still need to edit the user's `traits`. This only needs to be done once for every username, though. If you have server-based usernames instead, you can manage these via roles.

There's an open issue for editing users via UI: https://github.com/gravitational/teleport/issues/12225
